ui/redux/actions/collections.js

```diff
@@ -5,36 +5,34 @@ import Lbry from 'lbry';
 import { doClaimSearch, doAbandonClaim } from 'redux/actions/claims';
 import { makeSelectClaimForClaimId } from 'redux/selectors/claims';
 import {
-  makeSelectCollectionForId,
-  // makeSelectPublishedCollectionForId, // for "save" or "copy" action
-  makeSelectPublishedCollectionForId,
-  makeSelectUnpublishedCollectionForId,
-  makeSelectEditedCollectionForId,
+  selectCollectionForId,
+  // selectPublishedCollectionForId, // for "save" or "copy" action
+  selectPublishedCollectionForId,
+  selectUnpublishedCollectionForId,
+  selectEditedCollectionForId,
+  selectHasItemsInQueue,
 } from 'redux/selectors/collections';
 import * as COLS from 'constants/collections';
 
-const getTimestamp = () => {
-  return Math.floor(Date.now() / 1000);
-};
-
 const FETCH_BATCH_SIZE = 50;
 
-export const doLocalCollectionCreate = (
-  name: string,
-  collectionItems: Array<string>,
-  type: string,
-  sourceId: string
-) => (dispatch: Dispatch) => {
+export const doLocalCollectionCreate = (params: CollectionCreateParams, cb?: (id: any) => void) => (
+  dispatch: Dispatch
+) => {
+  const { items } = params;
+  const id = uuid();
+
+  if (cb) cb(id);
+
   return dispatch({
     type: ACTIONS.COLLECTION_NEW,
     data: {
       entry: {
-        id: uuid(), // start with a uuid, this becomes a claimId after publish
-        name: name,
-        updatedAt: getTimestamp(),
-        items: collectionItems || [],
-        sourceId: sourceId,
-        type: type,
+        id: id, // start with a uuid, this becomes a claimId after publish
+        createdAt: Date.now() / 1000,
+        updatedAt: Date.now() / 1000,
```

Comment:
- [ ] We need to retain the original code of including `Math.floor`, because the wallet data for `updatedAt` is an integer.
- [ ] While calling `Date.now()` is quick succession would result in the same value most of the time, I don't think this is guaranteed.  Many factors like browser implementation and execution delays could make the values drift.  Usually, just stash the value in a variable before the dispatch object is called.